File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/medal/UserMedalCreator.java
package com.depromeet.threedollar.domain.rds.user.domain.medal;

import org.jetbrains.annotations.NotNull;

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture;
import com.depromeet.threedollar.domain.rds.user.domain.user.User;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
@TestFixture
public class UserMedalCreator {

    @Builder(builderMethodName = "active")
    public static UserMedal createActive(
        @NotNull Medal medal,
        @NotNull User user
    ) {
        return UserMedal.builder()
            .medal(medal)
            .user(user)
            .status(UserMedalStatus.ACTIVE)
            .build();
    }

    @Builder(builderMethodName = "inActive")
    public static UserMedal createInActive(
        @NotNull Medal medal,
        @NotNull User user
    ) {
        return UserMedal.builder()
            .medal(medal)
            .user(user)
            .status(UserMedalStatus.IN_ACTIVE)
            .build();
    }

}

File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/medal/UserMedalCreator.kt
package com.depromeet.threedollar.domain.rds.user.domain.medal

import com.depromeet.threedollar.domain.rds.domain.userservice.medal.Medal
import com.depromeet.threedollar.domain.rds.domain.userservice.medal.UserMedal
import com.depromeet.threedollar.domain.rds.domain.userservice.medal.UserMedalStatus
import com.depromeet.threedollar.domain.rds.domain.userservice.user.User
import com.depromeet.threedollar.domain.rds.user.domain.TestFixture

@TestFixture
object UserMedalCreator {

    @JvmOverloads
    @JvmStatic
    fun create(
        medal: Medal,
        user: User,
        status: UserMedalStatus = UserMedalStatus.ACTIVE,
    ): UserMedal {
        return UserMedal.builder()
            .medal(medal)
            .user(user)
            .status(status)
            .build()
    }

}
